File: Frontend/interface/room.ts
import { Feature } from "./feature";
import { Hotel } from "./hotel";

export interface Room {
  id: number;
  title: string;
  description?: string;
  price: number; 
  guestCount: number;
  bedroomCount: number;
  bedCount: number;
  bathRoomCount: number;
  hotel: Hotel; 
  features: Feature[]; 
}

export interface RoomImage {
  id: number;       
  imageURL: string; 
  room: Room;      
  size: string;
}

export interface RoomResponse{
  id: number;
  hotelId: number;
  imageUrl: string; 
  title: string;
  hotelName: string;
  price: number
}

export interface RangePrice {
  minPrice: number;
  maxPrice: number;
}

export interface RoomDetail extends Omit<Room, "hotel">{
  hotelName: string;
  images: RoomImage[]
}

File: Backend/src/main/java/com/example/project_reactnative/model/entity/Room.java
package com.example.project_reactnative.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.HashSet;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String description;

    @Column(nullable = false)
    private BigDecimal price;

    private int guestCount;

    private int bedroomCount;

    private int bedCount;

    private int bathRoomCount;

    @ManyToOne
    @JoinColumn(name = "hotel_id", referencedColumnName = "id")
    private Hotel hotel;
}
